### ISSUE TITLE
Save File Optimizations

### ISSUE THREAD

**Issue by gheskett** [HackerN64/HackerSM64#358]
Save files should be optimized and allow for removal of backup saves via define.

**Comment by gheskett** [comment]
Related issue: #317

**Comment by gheskett** [comment]
Still unclear whether this is actually worth following through on, since it will break save trackers for saves not using Save File A

**Comment by arthurtilly** [comment]
this is something i do plan on doing (not as a define), i think we had a long discussion about this and planned to add padding to the save file to make it still compatible with save trackers by default but have the padding easy to remove

**Comment by gheskett** [comment]
Still think the best approach is to have one backup encompassing all four, though iirc? there was some other drawback that I can't remember anymore

**Comment by gheskett** [comment]
Another one actually is not writing backup saves means save file sizes _must_ be 8-byte aligned rather than just 4 like before

**Comment by arthurtilly** [comment]
adding one big backup with a weird implementation would ruin the entire point of this (that is, to make it trivial to increase the save file size significantly)

**Comment by gheskett** [comment]
It should still be easy if implemented properly, I don't see why it wouldn't. Just place 1 backup slot after the four main saves.

**Comment by arthurtilly** [comment]
i don't see how thats preferrable over the vanilla implementation and it doesnt solve the issue of being incompatible with star tracker software

**Comment by gheskett** [comment]
It's preferable over vanilla still because it's equivalent to 5 save slots instead of 8

**Comment by arthurtilly** [comment]
i guess, but having no backups means more code can be cleaned up, more space for save file expansion, more feasible to retain star tracker support and backups are not a very useful feature anyway

**Comment by gheskett** [comment]
Yeah I'm mostly just worried about save file corruption but I can probably live without it if it's for sure minor enough

**Comment by arthurtilly** [comment]
to be clear if backups are removed save file corruption would cause the save to be wiped automatically, it wouldnt just randomly give incorrect stars or anything

**Comment by gheskett** [comment]
Yeah, where I definitely draw the line is removing checksums

**Comment by arthurtilly** [comment]
removing checksums isnt even possible, the game relies on them to initialise the save data from EEPROM garbage
